src/transformers/models/llama/modeling_llama.py

```diff
@@ -278,9 +279,10 @@ def repeat_kv(hidden_states: torch.Tensor, n_rep: int) -> torch.Tensor:
 class LlamaAttention(nn.Module):
     """Multi-headed attention from 'Attention Is All You Need' paper"""
 
-    def __init__(self, config: LlamaConfig):
+    def __init__(self, config: LlamaConfig, layer_idx: int):
```

Comment:
```suggestion
    def __init__(self, config: LlamaConfig, layer_idx: Optional[int] = None):
```
Let's make sure we don't break backwards comp

Reply:
But on the other hand we really do need the layer index for the caching mechanism now
Just to be a bit safer and make it easier for users to transition, can we maybe do the following:
- https://github.com/huggingface/transformers/pull/26681/files#r1402008864
- https://github.com/huggingface/transformers/pull/26681/files#r1402007386